widget_weather.go

```diff
@@ -0,0 +1,136 @@
+		serializedCond = []uint8{127, 105, 135, 28, 137, 23, 142, 21, 143, 21, 144, 18, 148, 15, 150, 13, 152, 12, 152, 11, 153, 11, 153, 11, 153, 12, 152, 12, 151, 14, 149, 107, 129, 16, 129, 17, 131, 14, 131, 11, 129, 5, 129, 16, 129, 4, 129, 6, 131, 26, 131, 5, 129, 26, 131, 35, 129, 127, 110}
+	case "m": // partly cloudy
+		serializedCond = []uint8{127, 99, 130, 29, 130, 3, 130, 4, 129, 24, 136, 1, 131, 25, 138, 22, 130, 2, 138, 22, 144, 21, 145, 19, 144, 21, 141, 3, 134, 14, 141, 1, 137, 12, 153, 10, 155, 12, 152, 13, 153, 10, 155, 9, 130, 2, 152, 11, 154, 10, 154, 10, 154, 10, 154, 10, 154, 10, 153, 12, 152, 13, 150, 127, 93}
+	case "o": // sunny
+		serializedCond = []uint8{125, 129, 35, 130, 27, 129, 5, 131, 5, 129, 21, 131, 3, 132, 3, 130, 21, 132, 1, 133, 1, 132, 22, 142, 22, 142, 16, 131, 2, 144, 2, 131, 11, 152, 12, 151, 14, 150, 15, 148, 15, 149, 13, 153, 9, 157, 8, 156, 10, 151, 15, 148, 15, 149, 15, 150, 13, 152, 11, 154, 16, 144, 20, 143, 21, 142, 21, 132, 2, 132, 2, 131, 21, 130, 4, 131, 4, 130, 21, 129, 6, 130, 5, 129, 28, 130, 127, 34}
+	case "/!/": // thunder
+		serializedCond = []uint8{127, 106, 134, 28, 137, 23, 142, 21, 144, 20, 144, 19, 147, 15, 151, 12, 152, 12, 153, 10, 154, 10, 154, 10, 154, 11, 152, 12, 152, 13, 150, 15, 147, 24, 132, 31, 137, 27, 136, 32, 131, 32, 131, 33, 130, 33, 130, 33, 130, 127, 108}
+	case "!/", "*!*": // thunder rain
+		serializedCond = []uint8{127, 71, 132, 30, 136, 24, 141, 21, 143, 21, 144, 19, 146, 16, 150, 13, 152, 12, 152, 11, 154, 10, 154, 10, 154, 10, 154, 10, 153, 12, 152, 13, 150, 16, 146, 23, 133, 24, 130, 5, 135, 6, 130, 14, 131, 3, 136, 5, 131, 13, 131, 8, 131, 6, 130, 15, 129, 4, 129, 3, 131, 5, 129, 2, 129, 19, 130, 3, 130, 5, 130, 21, 131, 2, 130, 5, 131, 22, 129, 3, 129, 7, 129, 127, 101}
+	default:
+		serializedCond = []uint8{127, 63, 129, 11, 129, 23, 129, 11, 129, 23, 129, 10, 131, 22, 130, 9, 131, 22, 130, 8, 133, 20, 131, 7, 135, 19, 132, 7, 133, 19, 133, 8, 131, 20, 134, 7, 131, 19, 135, 8, 129, 5, 129, 14, 136, 7, 129, 5, 129, 13, 138, 12, 129, 12, 140, 10, 131, 10, 141, 10, 131, 11, 139, 11, 132, 11, 137, 11, 133, 12, 135, 11, 135, 12, 134, 10, 137, 11, 133, 11, 137, 12, 132, 12, 135, 13, 131, 14, 133, 14, 131, 15, 132, 15, 130, 15, 131, 16, 130, 15, 131, 16, 129, 17, 129, 17, 129, 17, 129, 35, 129, 127, 24}
+	}
+
+	offset := 5
+	index := 0
+	halfMax := 128
+	img := image.NewRGBA(image.Rect(0, 0, int(dev.Pixels), int(dev.Pixels)))
+	for _, pix := range serializedCond {
+		if int(pix) < halfMax {
+			index += int(pix)
+		} else {
+			for i := 0; i < int(pix)-halfMax; i++ {
+				img.Set(index%pixels+offset, index/pixels+offset, w.color)
+				index += 1
+			}
+		}
+	}
+
+	drawString(img,
+		img.Bounds(),
+		ttfFont,
+		temp,
+		dev.DPI,
+		14,
+		w.color,
+		image.Pt(-1, int(dev.Pixels)-7))
+
+	return w.render(dev, img)
+}
+
+func formatUnit(unit string) string {
+	switch unit {
+	case "f", "fahrenheit":
+		return "&u"
+	case "c", "celcius":
+		return "&u"
```

Comment:
This seems to return `&u` for both Fahrenheit & Celsius?

Reply:
Fixed that one dont know how that happened ...